pkg/util/util.go

```diff
@@ -232,29 +235,44 @@ func ParseMachinesYaml(file string) ([]*clusterv1.Machine, error) {
 	defer reader.Close()
 
 	decoder := yaml.NewYAMLOrJSONDecoder(reader, 32)
-	machineList, err := decodeMachineLists(decoder)
 
+	var (
+		bytes    [][]byte
+		machines = []clusterv1.Machine{}
+	)
+
+	// TODO: use the universal decoder instead of doing this.
+	bytes, err = decodeClusterV1Kinds(decoder, "MachineList")
```

Comment:
I believe using `:=` here can remove the var declaration above.

Reply:
I declared them upfront because I like making `if err != nil` statements terse, but then failed to actually do it 😅